apps/pkeyutl.c

```diff
@@ -40,6 +43,14 @@ static int do_raw_keyop(int pkey_op, EVP_MD_CTX *mctx,
                         int filesize, unsigned char *sig, int siglen,
                         unsigned char **out, size_t *poutlen);
 
+static int only_rawin(const EVP_PKEY *pkey)
+{
+    if (pkey == NULL)
+        return 0;
+    return EVP_PKEY_is_a(pkey, "ED25519")
+        || EVP_PKEY_is_a(pkey, "ED448");
```

Comment:
This, BTW, just isn't right.  The categorization should be made on capability; in this case, which API the implementation supports.  capability can be detected by trying to initialize the operation with `EVP_Digest{Whatever}Init()` and `EVP_PKEY_{whatever}_init()` (and `EVP_PKEY_{whatever}_message_init()` when we've landed that support)

Reply:
Yeah, I had asked about a smarter and more to-the-point way of doing this check.
Would be good if there was a general and simple-to-use EVP_PKEY API call for it.

Reply:
Agreed, it would be nice if there was a consistent capability query API for this sort of thing.